mars-cli/mars_lib/model.py

```diff
@@ -0,0 +1,296 @@
+from __future__ import annotations
+
+from enum import Enum
+from typing import List, Optional, Union
+
+from pydantic import BaseModel, Field, field_validator, ConfigDict
+from mars_lib.target_repo import TargetRepository, TARGET_REPO_KEY
+
+
+class IsaBase(BaseModel):
```

Comment:
I'm not an ISA or a Pydantic expert, but I'm wondering whether Pydantic gives us a way to extend the base ISA schemas to make "MARS-ISA-JSON" a reality. The idea here would be something like:

* ISA Python API contains the Pydantic models that exactly implement ISA JSON schema (if we can generate Pydantic classes from JSON schema automatically, this is hopefully not much overhead for their team - this code probably gives a good start).
* MARS extends those models with additional requirements - again I'm not an expert but something like [this gist](https://gist.github.com/shawnwall/6d0ae5f4a5b3e18d8403507616a3efb2) (from [this SO answer](https://stackoverflow.com/a/65907609)). This way our schema stays up-to-date with ISA, but we maintain our own additions to the schema.
* MARS can then validate using our additional requirements, and also use ISA-tools' own [object model](https://github.com/ISA-tools/isa-api/tree/master/isatools/model) to take advantage of things like GraphQL queries for filtering and so on.

(BTW none of affects this PR in particular, I'm just thinking/fantasizing out loud.)

Reply:
Yeah I agree with you.
There are python packages that generate pydantic models from json.
And pydantic models can spit out json schema as well.
Curious about what the rest thinks about it.

Reply:
how can we coordinate to reuse objects and functionality from the isa-api rather than re-implementing?

Note: the latest code now supports adding 'Comments' to Assays thus allowing to indicate which Repository to dispatch to.

Reply:
@proccaserra We would love to reuse the ISA API if it meets all our needs. But we were puzzled how we would for example validate, using ISA-tools, the loaded isa structure in a more stringent way than the ISA-JSON validator currently does? Because in the latest implementation we used, even an empty json was seen as a valid ISA JSON :/